pyxem/signals/indexation_results.py

```diff
@@ -147,6 +159,487 @@ def _get_second_best_phase(z):
+    for each navigation position, the signal contains the top n best matches in the form
+    of a nx4 array with columns [index,correlation, in-plane rotation, mirror(factor)]
+
+    The Simulation is saved in the metadata but can be accessed using the .simulation attribute.
+
+    Parameters
+    ----------
+    *args
+        See :class:`~hyperspy._signals.signal2d.Signal2D`.
+    **kwargs
+        See :class:`~hyperspy._signals.signal2d.Signal2D`
+    """
+
+    _signal_type = "orientation_map"
+
+    @property
+    def simulation(self):
+        return self.metadata.get_item("simulation")
+
+    @simulation.setter
+    def simulation(self, value):
+        self.metadata.set_item("simulation", value)
+
+    def deepcopy(self):
+        """Deepcopy the signal"""
+        self.simulation._phase_slider = None
+        self.simulation._rotation_slider = None
+        return super().deepcopy()
+
+    @property
+    def num_rows(self):
+        return self.axes_manager.signal_axes[1].size
+
+    def to_rotation(self, flatten=False):
+        """
+        Convert the orientation map to a set of `orix.Quaternion.Rotation` objects.
+        Returns
+        -------
+        """
+        if self._lazy:
+            warn("Computing the signal")
+            self.compute()
+        all_rotations = Rotation.stack(self.simulation.rotations).flatten()
+        rotations = self.map(
+            rotation_from_orientation_map,
+            rots=all_rotations.data,
+            inplace=False,
+            lazy_output=False,
+            output_signal_size=(self.num_rows, 4),
+            output_dtype=float,
+        )
+
+        rots = Rotation(rotations)
+        if flatten:
+            shape1 = np.prod(rots.shape[:-1])
+            rots = rots.reshape((shape1, rots.shape[-1]))
+        return rots
+
+    def to_phase_index(self):
+        """
+        Convert the orientation map to a set of phase ids
+
+        Returns
+        -------
+        np.ndarray or None
+            The phase ids for each pixel in the orientation map or None if the simulation
+            does not have multiple phases.
+        """
+        if self.simulation.has_multiple_phases:
+            sizes = np.cumsum([i.size for i in self.simulation.rotations])
+            return self.map(orientation2phase, sizes=sizes, inplace=False).data
+        else:
+            return None
+
+    def to_single_phase_orientations(self, **kwargs) -> Orientation:
+        """Convert the orientation map to an `Orientation`-object,
+        given a single-phase simulation.
+        """
+        if self.simulation.has_multiple_phases:
+            raise ValueError(
+                "Multiple phases found in simulation (use to_crystal_map instead)"
+            )
+
+        # Use the quaternion data from rotations to support 2D rotations,
+        # i.e. unique rotations for each navigation position
+        rotations = hs.signals.Signal2D(self.simulation.rotations.data)
+
```

Comment:
Is there a case where this is useful?  Currently this isn't really supported by the simulations. (All of the rotations are used so passing as a hyperspy signal doesn't do much). It just casts this to a numpy array and then passes that numpy array to all of the signals.

Reply:
No need to support it for now, we can add it without breaking the API later if we want